### ISSUE TITLE
Cloud sync is much slower than usual

### ISSUE THREAD

**Issue by netsonicyxf** [Tampermonkey/tampermonkey#1449]
## Expected Behavior
syncronization successful

## Actual Behavior
Show "Please wait" for ever

## Specifications

- Chromium: 98.0.4758.82 x64
- TM: 4.1.4
- OS: Win10 21H2

I enabled Tesla and tried Google drive, dropbox, onedrive, yedax, webdav,  
it always show "Please Wait"

**Comment by derjanb** [comment]
After pressing "Run now"? How many userscripts do you have installed? Can you please check the background page for running network requests (https://tmnk.net/faq#Q600 -> network tab)?

**Comment by WardenSlayer** [comment]
I am experiencing the same issue. I had to reinstall TM and lost my sync'd scripts that I had stored in Google docs. The fresh install won't sync and just hangs at please wait. I have no active tmnk.net connections. 

I also removed my app data from chrome and revoked its access to my drive to try and get it to start fresh. Still broken.

**Comment by netsonicyxf** [comment]
> After pressing "Run now"? How many userscripts do you have installed? Can you please check the background page for running network requests (https://tmnk.net/faq#Q600 -> network tab)?

I have 59 scripts. After I change the log level from "error" to "debug", the syncronization was successful. But switching back to the "error" level, the issue occurs again.

**Comment by LetMeDecay** [comment]
probably your server ssl issue, can not access your website https://tmnk.net/
![image](https://user-images.githubusercontent.com/7452372/159120841-19749e3f-a9f7-43a6-9094-767df5c2eacb.png)

**Comment by derjanb** [comment]
My bad. Please use http://tmnk.net/faq#Q600

**Comment by g0t4** [comment]
It appears as if no network requests are "wired up" to the Save button under the "Userscript Sync" section of settings. 

Furthermore, when I change other settings (ie log level) then the changes I've made to Userscript Sync are reverted back to defaults. And clicking "Run now" triggers the neverending "Please Wait" modal. 

This happens anytime I setutp a new install of TM and try to configure it. It appears settings sync (may) only work on existing installs prior to a recent release of TM that must be the culprit.

**Comment by derjanb** [comment]
> Furthermore, when I change other settings (ie log level) then the changes I've made to Userscript Sync are reverted back to defaults.

See #1438

**Comment by wang1212** [comment]
TM: 4.15 stable
SYSTEM: win7
BROWSER: chrome 101.0.4951.15 (dev)

---

I encountered the same problem, and in the settings, the settings related to synchronization cannot be saved, and the changes made after clicking the save button to refresh the page are lost.

**Comment by SmartManoj** [comment]
I have this bug for the last year.
Even 4.14.6145 is also not worked.
https://www.crx4chrome.com/crx/261990/

Could anyone suggest a last workable version?

**Comment by acbetter** [comment]
Same issue here from last year. current v4.16.6160, always wait...

![image](https://user-images.githubusercontent.com/13360124/163663246-5bb3124c-e624-4e5f-9737-cdc3654eac5b.png)

**Comment by Michaelvsk** [comment]
I did have the same issue with v4.16 and Microsoft Edge.
A **Factory Reset** fixed it for me - you'll find that option at bottom of settings page after setting _config mode_ to _Advanced_.

**Comment by netsonicyxf** [comment]
> I did have the same issue with v4.16 and Microsoft Edge. A **Factory Reset** fixed it for me - you'll find that option at bottom of settings page after setting _config mode_ to _Advanced_.

Factory Reset will delete all the scripts

**Comment by Michaelvsk** [comment]
> Factory Reset will delete all the scripts
Good to know. Anyway the _Utilities_ page provides an export to file/zip/cloud option to create a backup beforehand. ;)

**Comment by yutotakano** [comment]
Having the same issue. The "Please wait" dialogue simply runs for over 10 minutes with no sign of stopping anytime soon.

No network requests seem to be made at all when Saving, nor when Run Now is pressed. This is both with local storage and Google Drive. I suspected initially that the sync was too fast that it finished before the loading indicator appeared (thereby never dismissing it), but I don't want to risk the deletion of all my scripts to test this hypothesis out. Given I have 26 scripts I don't think it's a quick sync either.

If there's anything I can contribute via info, I'd love to.

Setting Log Level to "DEBUG", there is nothing at all on the Options page console. The Background page console displays "sync: schedule sync for run in 1 ms", but nothing after that.

**Comment by derjanb** [comment]
The latest BETA has a sync log field right after the sync settings in advanced config mode. (in review|[crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|[xpi](https://firefox.tampermonkey.net/firefox-current-beta.xpi))

**Comment by smu4242** [comment]
Same issue for me. `Settings -> Enable Userscript Sync` just hangs.

Workaround that helped in my case:
Under `Utilities` there is `Cloud` and then `Export`. This requires to manually do `Show Backups` on the other device, but that worked without an issue.
There's also quite a few other options there like `Zip`, `File`, and `Textarea`, which I have not tried.

I didn't see any of these at first, because I just didn't expect there to be yet another possibility to do a "sync". I understand that "sync" and "backup" are different things, but I would suggest to move these all into the same dialog, as they are closely related.

**Comment by amartini** [comment]
Had this after doing a sync with an old backup. After Factory Reset it worked.

**Comment by tkkcc** [comment]
after `factory reset`, `browser sync` mode works for the first 3 scripts, and then hanging. switch to `google drive` mode works.

**Comment by vfonic** [comment]
> Same issue for me. `Settings -> Enable Userscript Sync` just hangs.

For me the initial sync takes about 1-5 seconds per script. I have ~100 scripts, so that's 1min-8min to do the initial sync. After that, it's much faster.

> Workaround that helped in my case:
> Under `Utilities` there is `Cloud` and then `Export`. This requires to manually do `Show Backups` on the other device, but that worked without an issue.

I used this to get the initial backup restored on a new machine. After this I enabled Userscript sync.

Anyone knows how often the Userscript sync is run?
I don't understand from this UI whether it's realted to above mentioned items or it's a completely different schedule or just manual? cc @derjanb 

![Image](https://github.com/user-attachments/assets/147e1e89-585e-4231-856b-29752c159b51)
